pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -327,32 +328,98 @@ func newDeploymentSettingsConfigureCmd() *cobra.Command {
 	return cmd
 }
 
+func newRepoLookup(repoRoot string) (repoLookup, error) {
+	if repoRoot != "" {
+
+		repo, err := git.PlainOpen(repoRoot)
+		if err != nil {
+			return nil, err
+		}
+
+		h, err := repo.Head()
+		if err != nil {
+			return nil, err
+		}
+
+		return &repoLookupImpl{
+			RepoRoot: repoRoot,
+			Repo:     repo,
+			Head:     h,
+		}, nil
+	}
+	return &noRepoLookupImpl{}, nil
+}
+
+type repoLookup interface {
+	GetRootDirectory(wd string) (string, error)
+	GetBranchName() string
+	RemoteURL() (string, error)
+}
+type repoLookupImpl struct {
+	RepoRoot string
+	Repo     *git.Repository
+	Head     *plumbing.Reference
+}
+
+func (r *repoLookupImpl) GetRootDirectory(wd string) (string, error) {
+	return filepath.Rel(r.RepoRoot, wd)
+}
+
+func (r *repoLookupImpl) GetBranchName() string {
+	return r.Head.Name().String()
```

Comment:
This probably needs a nil check before dereferencing `Head`.

Reply:
I believe it is unlikely if it doesnt return an error but wont hurt to be defensive, done